bson/src/test/scala/org/mongodb/scala/bson/codecs/MacrosSpec.scala

```diff
@@ -141,6 +145,11 @@ class MacrosSpec extends FlatSpec with Matchers {
     decode(registry.get(classOf[OptionalValue]), buffer) should equal(OptionalValue("Bob", None))
   }
 
+  it should "be able to round trip ignored optional values" in {
+    roundTrip(OptionalValueIgnoreNone("Bob", None), """{name: "Bob"}""", classOf[OptionalValueIgnoreNone])
```

Comment:
`OptionalValueIgnoreNone("Bob", null)` should be left as is, it is pure scala and a user error to try and create that case class instance.